### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by NathanBaulch**
**What this PR does / why we need it**:

Just thought I'd contribute some typo fixes that I stumbled on. Nothing controversial (hopefully), just 23 simple fixes.

Use the following command to get a quick and dirty summary of the specific corrections made:
```shell
git diff HEAD^! --word-diff-regex='\w+' -U0 \
  | grep -E '\[\-.*\-\]\{\+.*\+\}' \
  | sed -r 's/.*\[\-(.*)\-\]\{\+(.*)\+\}.*/\1 \2/' \
  | sort | uniq -c | sort -n
```

FWIW, the typos are:
* lenght (4)
* suplied (4)
* alwaty (2)
* chechk (1)
* addes (1)
* appliyed (1)
* assigment (1)
* automattic (1)
* complient (1)
* cosumed (1)
* exists (1)
* friendy (1)
* partioned (1)
* suberesource (1)
* supercede (1)
* unparseable (1)

### CODE REVIEW

**Review by mattfarina — APPROVED** [review]

**Review by robertsirc — APPROVED** [review]
LGTM

File: pkg/action/install.go [hunk 1]
```diff
@@ -631,7 +631,7 @@ func createOrOpenFile(filename string, append bool) (*os.File, error) {
 	return os.Create(filename)
 }
 
-// check if the directory exists to create file. creates if don't exists
+// check if the directory exists to create file. creates if don't exist
```
Comment (joejulian) [inline pkg/action/install.go:634]:
```suggestion
// check if the directory exists to create file. creates if does't exist
```

Comment (NathanBaulch) [reply pkg/action/install.go]: I've thrown in a few grammar corrections including this one 👍

**Review by yardenshoham — APPROVED** [review]

File: cmd/helm/load_plugins.go [hunk 2]
```diff
@@ -286,7 +286,7 @@ func addPluginCommands(plugin *plugin.Plugin, baseCmd *cobra.Command, cmds *plug
 					f.BoolP(longs[i], shorts[i], false, "")
 				} else {
 					// Create a long flag with the same name as the short flag.
-					// Not a perfect solution, but its better than ignoring the extra short flags.
+					// Not a perfect solution, but it's better than ignoring the extra short flags.
```
Comment (TerryHowe) [inline cmd/helm/load_plugins.go:289]: I generally think it is better to avoid contractions

**Review by TerryHowe — APPROVED** [review]
/lgtm

File: cmd/helm/load_plugins.go (hunk 2 above)
Comment (NathanBaulch) [reply cmd/helm/load_plugins.go:289]: I don't mind either way. The repo contains a mixture of both (21 instances of "it's") so I went with the simpler addition of an apostrophe.

**Review by mattfarina — APPROVED** [review]
@joejulian what do you think now?

**Review by sabre1041 — APPROVED** [review]
LGTM

**Pull request merged**